app/src/util/tools.ts

```diff
@@ -472,3 +472,11 @@ export const onChangeMarketCurrency = (
     }
   }
 }
+
+export const calcPrediction = (probability: string, lowerBound: BigNumber, upperBound: BigNumber, decimals: number) => {
```

Comment:
Yeah, I had considered this initially, but since the probability string is greater than 18 decimals we can actually get more precision using Number types and it works out well since we want a number as the return value.
That being said I actually forgot to set the precision for `formatBigNumber`, so that is now properly precise.